lib/inertia_rails/renderer.rb

```diff
@@ -112,5 +115,32 @@ def resolve_component(component)
 
       configuration.component_path_resolver(path: controller.controller_path, action: controller.action_name)
     end
+
+    def validate_partial_props(merged_props)
+      return unless rendering_partial_component?
+
+      unrequested_props = merged_props.select do |key, prop|
+        !key.in?(partial_keys)
+      end
+
+      props_that_will_unecessarily_compute = unrequested_props.select do |key, prop|
```

Comment:
genuine question: Is there a valid use-case for a prop here that is callable, but evaluates every time?

Reply:
I'm not sure I understand the question. Props that are callable get called (or skipped) after this method runs.

This method isn't doing anything more fancy than checking to see if there are any unrequested props that were defined as values (and it short circuits unless we're partial reloading). The filtering doesn't mutate anything or get used in the actual computing/rendering.